crates/primitives/src/block.rs

```diff
@@ -1,18 +1,19 @@
 use crate::{Header, SealedHeader, TransactionSigned, H256};
-use reth_codecs::derive_arbitrary;
+use reth_codecs::{derive_arbitrary, main_codec, Compact};
 use reth_rlp::{Decodable, DecodeError, Encodable, RlpDecodable, RlpEncodable};
 use serde::{Deserialize, Serialize};
 use std::ops::Deref;
 
 /// Ethereum full block.
+#[main_codec]
```

Comment:
I guess this was not required before because we'd only write a block as Bodies + Transactions, whereas now we're actually writing the full block for persistence across crashes in the fork tracking?